measures/utility_bills/measure.rb

```diff
@@ -60,6 +60,95 @@ def energyPlusOutputRequests(runner, user_arguments)
     return result
   end
 
+  # return filtered electricity utility rates
+  def filter_datapoints_with_median_bounds(runner, rate_results, list_of_labels_to_remove, create_list)
+    # initialize filtered electricity rate variable
+    elec_bills = {}
+
+    # report bill breakdowns: demand charge
+    bills_sorted = rate_results.values.reject(&:zero?).sort
+
+    rate_results.each do |rate_label, bill|
+      # if there are no bills left after removing zeros, set to 0
+      if bills_sorted.empty?
+        elec_bills[rate_label] = 0
+        next
+      end
+
+      # if the rate label is considered outlier from total bill stand point, don't include
+      if list_of_labels_to_remove.include?(rate_label)
+        next
+      end
+
+      # get median
+      mid = bills_sorted.length / 2
+      median_bill = bills_sorted.length.odd? ? bills_sorted[mid] : (bills_sorted[mid - 1] + bills_sorted[mid]) / 2.0
+
+      # skip outliers
+      if bill < 0.25 * median_bill
+        if create_list
+          list_of_labels_to_remove << rate_label
+        end
+        runner.registerInfo("Removing #{rate_label}, because bill #{bill} < 0.25 x median #{median_bill}")
+        next
+      elsif bill > 2.0 * median_bill
+        if create_list
+          list_of_labels_to_remove << rate_label
+        end
+        runner.registerInfo("Removing #{rate_label}, because bill #{bill} > 2.0 x median #{median_bill}")
+        next
+      end
+
+      # include the bill result in bill result statistics
+      elec_bills[rate_label] = bill
+    end
+
+    return elec_bills, list_of_labels_to_remove.uniq
```

Comment:
Would it be possible to have duplicated labels in the list?

Reply:
yeah that was actually what was happening with my testing. so just put a .uniq and called it done.